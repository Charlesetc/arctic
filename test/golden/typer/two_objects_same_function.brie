define sleep : x [
        x.sleep
        x
]
define main [
        define x < sleep = 2, hi = 3 > 
        define y < sleep = "cool", there = 4 >
        (sleep x).hi
        (sleep y).there
]

