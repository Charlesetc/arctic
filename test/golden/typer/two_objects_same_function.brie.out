root[(ident(define):: ident(sleep):: block[ident(x)::][(ident(x)::.sleep::)::, (ident(x)::)::]::Function)::, (ident(define):: ident(main):: block[][(ident(define):: ident(x):: <sleep = (ident(2)::Integer)::Integer , hi = (ident(3)::Integer)::Integer>::<sleep = Integer, hi = Integer>)::, (ident(define):: ident(y):: <sleep = (string(cool)::String)::String , there = (ident(4)::Integer)::Integer>::<sleep = String, there = Integer>)::, ((ident(sleep)::Function ident(x)::<sleep = Integer, hi = Integer>)::<sleep = Integer, hi = Integer>.hi::Integer)::Integer, ((ident(sleep)::Function ident(y)::<sleep = String, there = Integer>)::<sleep = String, there = Integer>.there::Integer)::Integer]::Function)::]::